packages/calling-stateful-client/src/CallClientState.ts

```diff
@@ -187,6 +189,31 @@ export type RaisedHandState = {
   raisedHandOrderPosition: number;
 };
 
+/* @conditional-compile-remove(reaction) */
+/**
+ * State only version of {@link @azure/communication-calling#ReactionCallFeature}. {@link StatefulCallClient} will
+ * automatically listen for reaction on the call and update the state exposed by {@link StatefulCallClient} accordingly.
+ * 
+ * @public
+ */
+export interface ReactionCallFeatureState {
+  isEnable: boolean;
+  localParticipantReactionPayload?: ReactionState;
+}
+
+/* @conditional-compile-remove(reaction) */
+/**
+ * State only version of {@link @azure/communication-calling#Call.Reaction.ReactionEventPayload} with UI helper props.
+ * Reation state with a comparative timestamp which helps UI to decide to render the reaction accordingly.
+ * 
+ * @public
+ */
+export type ReactionState = {
+  shouldRender: boolean;
```

Comment:
Avoid this property at the stateful layer. The stateful layer isn't tied to rendering logic and is just a reflection of the active call.
The composite or component will need to handle the logic if if the reaction should be played or not

Reply:
We could replace it with - `isAllowedByMeetingPolicy: boolean;`